### PR TITLE
Fix HMAC support

### LINKED ISSUE #307: Sign with HMAC algo is keypair bound
I'm using the bundle with lcobucci provider, his library offers an HMAC sign feature using only a secret, the bundle instead seems to not support it requiring a private/public certificate.

in \Lexik\Bundle\JWTAuthenticationBundle\Services\JWSProvider\LcobucciJWSProvider::create:66
```
new Key($this->keyLoader->loadKey('private'), $this->keyLoader->getPassphrase());

// can be too

new Key('secret')
```

and in \Lexik\Bundle\JWTAuthenticationBundle\Services\JWSProvider\LcobucciJWSProvider:91
```
$jws->verify($this->signer, $this->keyLoader->loadKey('public')) && $jws->validate(new ValidationData());

// can be too

$jws->verify($this->signer, '--> secret <--') && $jws->validate(new ValidationData());
```

do you plan to support it ? I can try to prepare a pr if it's ok for you

### LINKED ISSUE #340: Allow passing RAW keys directly in config (from "private_key_path" to "private_key")
Changing `public_key_path`/`private_key_path` to `public_key`/`private_key`, allowing to pass the key content directly as well as its file path.
This would allow storing keys as environment variables, btw removing the need for git-ignoring these keys if stored in the project itself (fixing https://github.com/symfony/recipes/pull/89).

This also relates to one thing I'd like to get in: allow using symmetric key algorithms (shared key,   https://github.com/lexik/LexikJWTAuthenticationBundle/projects/1#card-2657179)

### PR DESCRIPTION
**Pull request by chalasr**
- [x] Config
- [x] Encoders

Fixes #307 
Fixes #340

### CODE REVIEW

**Comment by railto** [comment]
Any update as to when this could be implemented?

**Comment by chalasr** [comment]
@railto This will be released by the end of the month, some cleanup needed.

**Comment by piteur** [comment]
@chalasr - Can't wait for this to be implemented. It'll be easier to deploy with docker thanks to this PR. No more volume to store only the keys...
Thanks for your work !

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -25,12 +25,23 @@ public function getConfigTreeBuilder()
             ->addDefaultsIfNotSet()
             ->children()
                 ->scalarNode('private_key_path')
+                    ->setDeprecated('The "%path%.%node%" configuration key is deprecated since version 2.5. Use "%path%.secret_key%" instead.')
```
Comment (Oliboy50) [inline DependencyInjection/Configuration.php]: I think the new configuration key should contain 2 or 4 `%` symbols, not 3, right?

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -25,12 +25,23 @@ public function getConfigTreeBuilder()
...
                     ->defaultNull()
                 ->end()
                 ->scalarNode('public_key_path')
+                    ->setDeprecated('The "%path%.%node%" configuration key is deprecated since version 2.5. Use "%path%.public_key%" instead.')
```
Comment (Oliboy50) [inline DependencyInjection/Configuration.php]: same

File: DependencyInjection/Configuration.php (hunk 1 above)
Comment (chalasr) [reply DependencyInjection/Configuration.php]: good catch

**Comment by MasseElch** [comment]
@chalasr What is the progress here? Do you need any help?

**Comment by chalasr** [comment]
Sorry for being latish guys, this is ready and will be merged today (not now in case one wants to give it a last review).

File: DependencyInjection/LexikJWTAuthenticationExtension.php [hunk 3]
```diff
@@ -52,8 +49,20 @@ public function load(array $configs, ContainerBuilder $container)
...
+            $config['secret_key'] = $config['private_key_path'];
+            $container->setParameter('lexik_jwt_authentication.private_key_path', $config['secret_key']);
+        }
+
+        if (isset($config['public_key_path'])) {
+            $config['public_key'] = $config['public_key_path'];
+            $container->setParameter('lexik_jwt_authentication.public_key_path', $config['public_key']);
+        }
+
+        if (empty($config['public_key']) && empty(['secret_key'])) {
```
[the remark this answers is not in the document]
Comment (chalasr) [reply DependencyInjection/LexikJWTAuthenticationExtension.php]: nice catch! thanks

**Pull request merged**